src/components/showHide/showHide.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @ngdoc module
+ * @name material.components.showHide
+ */
+
+angular.module('material.components.showHide', [
+  'material.core'
+])
+  .directive('ngShow', createDirective('ngShow', true))
+  .directive('ngHide', createDirective('ngHide', false));
+
+
+// Add additional handlers to ng-show and ng-hide that notify directives
+// contained within that they should recompute their size.
+function createDirective(name, targetValue) {
+  return ['$mdUtil', function($mdUtil) {
+    return {
+      restrict: 'A',
+      multiElement: true,
+      link: function($scope, $element, $attr) {
+        $scope.$watch($attr[name], function(value) {
+          if (!!value === targetValue) {
+            $scope.$broadcast('$md-resize');
+            $mdUtil.dom.animator.waitTransitionEnd($element).then(function() {
```

Comment:
Yeah, my concern was that not every element has transitions, and 3s is longer than I'd like to wait in those cases.